libs/github_utils.py

```diff
@@ -0,0 +1,85 @@
+from typing import Tuple
+import shutil
+import tempfile
+import io
+import zipfile
+import pathlib
+import logging
+import requests
+
+
+_logger = logging.getLogger(__name__)
+
+REPO = "covid-projections/covid-data-model"
+
+REPO_GIT_URL = f"https://api.github.com/repos/{REPO}"
+# GitHub assigned ID for "Build & Publish API artifacts" workflow.
+WORKFLOW_ID = "988804"
+
+
+def _get_artifact_zip_url(run_number: int = None, latest: bool = True) -> Tuple[str, int]:
+
+    workflow_runs_url = f"{REPO_GIT_URL}/actions/workflows/{WORKFLOW_ID}/runs?status=completed"
+
+    runs = requests.get(workflow_runs_url).json()["workflow_runs"]
+    if latest:
+        # Workflows are returned latest first.
+        run = runs[0]
+    else:
+        # TODO: Only requests the first page of workflows, if run id is not from the
+        # 30 most recent runs, it will not be able to be found.
+        for run in runs:
+            if run["run_number"] == run_number:
+                break
+        else:
+            raise Exception("Could not find workflow with run number {run_number}.")
```

Comment:
add something in this message about only supporting recent runs so users don't need to dig up the TODO in the source.